core/src/main/java/io/github/tcdl/Collector.java

```diff
@@ -96,13 +97,13 @@ public boolean isAwaitingResponses() {
     public void listenForResponses(String topic, Message requestMessage) {
         this.topic = topic;
         this.requestMessage = requestMessage;
-        channelManager.subscribe(this.topic, this);
+        collectorSubscriber.registerCollector(topic, this);
+        channelManager.subscribe(this.topic, collectorSubscriber);
```

Comment:
no this only the name, actually subscribe is allias for "old" findOrCreateStuff - first time we create Consumer we do subscribe. All next calls will cause nothing